doc/collision_detection.md

```diff
@@ -0,0 +1,212 @@
+intersected. Note that the `onCollision` method will be called on both collidable components if they
+have both implemented the `onCollision` method.
+
+If you want to check collisions with the screen edges, as we do in the example above, you can use
+the predefined [ScreenCollidable](#ScreenCollidable) class and since that one aldo is a `Collidable`
+you can implement your own `onCollision` method for that class if needed.
+
+### HasCollidables
+If you want to use this collision detection in your game you have to add the `HasCollidables` mixin
+to your game so that the game knows that it should keep track of which components that can collide.
+
+Example:
+```dart
+class MyGame extends BaseGame with HasCollidables {
+  ...
+}
+```
+
+Now when you add your `Collidable` components to your game they will automatically be checked for
+collisions.
+
+### ScreenCollidable
+`ScreenCollidable` is not a mixin, but a pre-made collidable component which represents the edges of
+your game/screen. If you add a `ScreenCollidable` to your game your other collidables will be
+notified when they collide with the edges. It doesn't take any arguments, it only depends on the
+`size` of the game that it is added to. To add it you can just do `add(ScreenCollidable())` in
+your game, if you don't want the `ScreenCollidable` itself to be notified when something collides
+with it. If you so want it to be notified, then you need to extend it and implement the
+`collisionCallback` method for it.
+
+## Shapes
+### Shape
+A Shape is the base class for representing a scalable geometrical shape. The shapes have different
+ways of defining how they look, but they all have a size and angle that can be modified and the
+shape definition will scale or rotate the shape accordingly.
+
+There are currently three shapes: [Polygon}(#Polygon), [Rectangle](#Rectangle) and
+[Cirlce](#Circle).
+
+### HitboxShape
+A HitboxShape is a Shape defined from the center position of the component that it is attached to
+and it has the same bounding size and angle as the component. You can set `localPosition` to have
+the position of the shape deviate from the center of the component. A HitboxShape is the type of
+shape that you add to your Hitbox, or Collidable. Usually these types of shapes are the only ones
+that you need to use.
+
+#### HitboxPolygon
+It should be noted that if you want to use collision detection or `containsPoint` on the `Polygon`,
+the polygon needs to be convex. So always use convex polygons or you will most likely run into
+problems if you don't really know what you are doing. It should also be noted that you should always
+define the vertices in your polygon in a clockwise order.
+
+In comparision to the normal `Polygon`, there is only one way to create a `HitboxPolygon`, the only
+mandatory argument is a list of `Vector2` which defines how your polygon should look, but not the
+size or position for it (since that sill be defined by the component that you attach it to).
```

Comment:
I would explicitly say here that the reference is a coordinate system where the center of the component is the origin and the coordinates go from -1 to 1 matching the `size` of the component.

Reply:
ah I see you do that below, sorry!